src/pages/modules/ModulesPage.slice.ts

```diff
@@ -0,0 +1,138 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+  IModuleState,
+  IModulesGETFailure,
+  IModulesPUT,
+  IModulesPUTFailure,
+  IModule,
+  AlertType,
+} from "../../shared/types";
+import { AlertMsg } from "./config";
+
+const initialModuleState: IModuleState = {
+  modules: [],
+  isLoading: true,
+  feedback: {
+    message: "",
+    display: false,
+    type: AlertType.info,
+  },
+};
+
+export function getInitialModuleState(): IModuleState {
+  return { ...initialModuleState };
+}
+
+export const moduleSlice = createSlice({
+  name: "modules",
+  initialState: getInitialModuleState(),
+  reducers: {
+    /* GET Request Modules */
+    requestModules: (state, action: PayloadAction<void>) => {
+      return { ...state, isLoading: true };
+    },
+
+    // action.type = "modules/requestModulesSuccess"
+    requestModulesSuccess: (state, action: PayloadAction<IModule[]>) => {
+      console.log("getting modules success");
+      console.log(action);
+      return {
+        ...state,
+        modules: action.payload,
+        feedback: {
```

Comment:
nit: do we have to tell the user when getting the modules explicitly? won't they be able to tell by seeing the modules?

Reply:
Of course we do not have to, but I like giving them feedback, like when adding a module, even when they can see the module they added.

Also, you can select a group of lines when adding a comment so that its easier for me to tell what function you are referring to.